server/db/models/player.js

```diff
@@ -148,204 +89,183 @@ playerSchema.methods.setCardinal = function(row, col) {
 };
 
 
-playerSchema.methods.getOpponents = function(){
-  return this.model('Player')
-  .find({game: this.game})
-  .where({_id :{$ne: this._id}})
-  .bind(this);
-};
-
 playerSchema.methods.boardMove = function (bearing) {
   // call when self.bearing is not relevant
   var newCol = this.position[0];
   var newRow = this.position[1];
   newCol += bearing[0];
   newRow += bearing[1];
-  return
-  this.set('position', [newCol, newRow]);
+  // check if move is possible
+  if (newCol < 0 || newRow < 0 || newCol > 11 || newRow > 15) return this.loseLife();
+  else {
+    this.set('position', [newCol, newRow]);
+  }
 };
 
-playerSchema.methods.pushPlayer = function (){
-  return this.getOpponents().bind(this)
-  .then(function(opponents){
-    return Promise.map(opponents, function(o){
-      // if
-    })
-  })
+
+playerSchema.methods.loseLife = function() {
+  this.livesRemaining--;
+  if (this.livesRemaining === 0) return this.killPlayer();
+  else {
+    this.set('position', this.dock);
+  }
 };
 
-playerSchema.methods.checkMove = function(bearing) {
-  var key = 'edge' + bearing[2];
-  return this.findMyTile()
-  .then(function(tile) {
-    if (tile[key] === null) return true;
-    else return false
-  })
+
+playerSchema.methods.killPlayer = function() {
+  this.position = null;
+};
+
+
+function findOppositeBearing(bearing) {
+  if (bearing == [-1,0,'N']) return [1,0,'S'];
+  if (bearing == [0,1,'E']) return [0,-1,'W'];
+  if (bearing == [1,0,'S']) return [-1,0,'N'];
+  if (bearing == [0,-1,'W']) return [0,1,'E'];
+}
+
```

Comment:
deleted this helper function and modified checkMove function to utilize a hash to check walls attempting to exit and enter